pkgs/by-name/gr/grist-core/package.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, fetchYarnDeps
+, yarn
+, nodejs
+, prefetch-yarn-deps
+, fixup-yarn-lock
+, nodePackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "grist-core";
+  version = "1.1.13";
+
+  src = fetchFromGitHub {
+    owner = "gristlabs";
+    repo = "grist-core";
+    rev = "v${version}";
+    hash = "sha256-lLXgTVhztFGnfrMxks05T8dfE6CH4p/0j8lPieBRGTY=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-cbkykydPfIrlCxaLfdIFseBuNVUtIqPpLR2J3LTFQl4=";
+  };
+
+  nativeBuildInputs = with nodePackages; [
+    yarn
+    nodejs
+    prefetch-yarn-deps
+    fixup-yarn-lock
+    node-gyp-build
+    node-pre-gyp
+  ];
+
+  propagatedBuildInputs = with python3.pkgs;
+    [
+      astroid
+      asttokens
+      chardet
+      et-xmlfile
+      executing
+      friendly-traceback
+      iso8601
+      lazy-object-proxy
+      openpyxl
+      phonenumbers
+      pure-eval
+      python-dateutil
+      roman
+      six
+      sortedcontainers
+      stack-data
+      typing-extensions
+      unittest-xml-reporting
+      wrapt
+    ];
+
+  postPatch = ''
+    rm .yarnrc
+  '';
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+    yarn config --offline set yarn-offline-mirror ${offlineCache}
+    fixup-yarn-lock yarn.lock
+
+    mkdir -p "$HOME/.node-gyp/${nodejs.version}"
+    echo 9 >"$HOME/.node-gyp/${nodejs.version}/installVersion"
+    ln -sfv "${nodejs}/include" "$HOME/.node-gyp/${nodejs.version}"
+    export npm_config_nodedir=${nodejs}
+
+    yarn --offline --frozen-lockfile --ignore-platform --ignore-engines --no-progress --non-interactive install
+    patchShebangs node_modules
+    patchShebangs buildtools
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    yarn --offline run build:prod
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    cp -r . $out
```

Comment:
I don't think that copying everything is necessary, certainly not node_modules as the js is only for running the client in the web browser, https://github.com/search?q=repo%3Agristlabs%2Fgrist-core+bundle.js&type=code and https://github.com/gristlabs/grist-core/blob/main/buildtools/webpack.config.js semm to indicate that all the js outputs will be put in the `static` folder and are self-contained

Reply:
grist needs node_modules to start. I tried to copy everything except node_modules, but when I started grist-core, it didn't find a module :/